### ISSUE TITLE
Smartmatch dies with a sub and a hash

### ISSUE THREAD

**Issue by FCO** [rakudo/rakudo#2560]
https://colabti.org/irclogger/irclogger_log/perl6?date=2018-12-22#l32

**Comment by lizmat** [comment]
Golfed to:

    Map.new.ACCEPTS(Any);

or:

    Map.new.ACCEPTS(Positional);

It tries to do an `any` on the parameter, and then feeds that `Junction` to `.EXISTS-KEY`.  And somewhere in the auto-threading we got the explosion.

**Comment by vrurg** [comment]
I've  stumbled upon this issue just today and can't help myself but express my wonder how come it never caused more problems in other code? It is not such that it explodes somewhere in autothreading, but it's the corresponding `ACCEPTS` candidates signatures which make thing go "boom":

```raku
    multi method ACCEPTS(Map:D: Any $topic --> Bool:D) {
        self.EXISTS-KEY($topic.any);
    }
```

Basically, this will **never** work because we explicitly create a `Junction`, `EXISTS-KEY` autothreads over it and produces a `Junction` as a return value, which is then gets rejected by `Bool:D` constraint!

I will try to remember to get this fixed after rakudo/rakudo#4653 is merged.
